lib/src/builder/util.dart

```diff
@@ -0,0 +1,65 @@
+import 'dart:mirrors';
+
+import 'package:analyzer/analyzer.dart';
+import 'package:path/path.dart' as p;
+import 'package:build/build.dart' show AssetId;
+import 'package:source_span/source_span.dart';
+
+const outputExtension = '.over_react.g.dart';
+const generatedPrefix = r'_$';
+const privatePrefix = r'_';
+const publicGeneratedPrefix = r'$';
+
+
+String getName(Type type) {
+  return MirrorSystem.getName(reflectType(type).simpleName);
+}
+
+/// Converts [id] to a "package:" URI.
+///
+/// This will return a schemeless URI if [id] doesn't represent a library in
+/// `lib/`.
+Uri idToPackageUri(AssetId id) {
+  if (!id.path.startsWith('lib/')) {
+    return new Uri(path: id.path);
+  }
+
+  return new Uri(scheme: 'package',
+      path: p.url.join(id.package, id.path.replaceFirst('lib/', '')));
+}
+
+/// Returns a string representing a [TypeParameterList], but type bounds removed.
+///
+/// Example:
+///   Input:
+///     '<T extends Iterable, U>' //TypeParameterList.toSource()
+///   Output:
+///     '<T, U>'
+String removeBoundsFromTypeParameters(TypeParameterList typeParameters) {
+  return typeParameters != null ? (StringBuffer()
+    ..write('${typeParameters.leftBracket}')..write(
+        typeParameters.typeParameters.map((t) => t.name.name).join(
+            ', '))..write('${typeParameters.rightBracket}'))
+      .toString()
+      : '';
```

Comment:
This is one of those blocks of code that ... when you see it... you wonder - WITAF is dart format doing?

Like... how could this _possibly_ be the "canonical" way to format this code?